source/bdDetect.py

```diff
@@ -545,267 +559,11 @@ def initialize():
-		any(
-			m.id.startswith(prefix)
-			for prefix in (
-				"F14", "Focus 14 BT",
-				"Focus 40 BT",
-				"Focus 80 BT",
-			)
-		)
-	))
-
-	# handyTech
-	addUsbDevices("handyTech", KEY_SERIAL, {
-		"VID_0403&PID_6001",  # FTDI chip
-		"VID_0921&PID_1200",  # GoHubs chip
-	})
-
-	# Newer Handy Tech displays have a native HID processor
-	addUsbDevices("handyTech", KEY_HID, {
-		"VID_1FE4&PID_0054",  # Active Braille
-		"VID_1FE4&PID_0055",  # Connect Braille
-		"VID_1FE4&PID_0061",  # Actilino
-		"VID_1FE4&PID_0064",  # Active Star 40
-		"VID_1FE4&PID_0081",  # Basic Braille 16
-		"VID_1FE4&PID_0082",  # Basic Braille 20
-		"VID_1FE4&PID_0083",  # Basic Braille 32
-		"VID_1FE4&PID_0084",  # Basic Braille 40
-		"VID_1FE4&PID_008A",  # Basic Braille 48
-		"VID_1FE4&PID_0086",  # Basic Braille 64
-		"VID_1FE4&PID_0087",  # Basic Braille 80
-		"VID_1FE4&PID_008B",  # Basic Braille 160
-		"VID_1FE4&PID_008C",  # Basic Braille 84
-		"VID_1FE4&PID_0093",  # Basic Braille Plus 32
-		"VID_1FE4&PID_0094",  # Basic Braille Plus 40
-		"VID_1FE4&PID_00A4",  # Activator
-	})
-
-	# Some older HT displays use a HID converter and an internal serial interface
-	addUsbDevices("handyTech", KEY_HID, {
-		"VID_1FE4&PID_0003",  # USB-HID adapter
-		"VID_1FE4&PID_0074",  # Braille Star 40
-		"VID_1FE4&PID_0044",  # Easy Braille
-	})
-
-	addBluetoothDevices("handyTech", lambda m: any(m.id.startswith(prefix) for prefix in (
-		"Actilino AL",
-		"Active Braille AB",
-		"Active Star AS",
-		"Basic Braille BB",
-		"Basic Braille Plus BP",
-		"Braille Star 40 BS",
-		"Braillino BL",
-		"Braille Wave BW",
-		"Easy Braille EBR",
-		"Activator AC",
-	)))
-
-	# hims
-	# Bulk devices
-	addUsbDevices("hims", KEY_CUSTOM, {
-		"VID_045E&PID_930A",  # Braille Sense & Smart Beetle
-		"VID_045E&PID_930B",  # Braille EDGE 40
-	})
-
-	# Sync Braille, serial device
-	addUsbDevices("hims", KEY_SERIAL, {
-		"VID_0403&PID_6001",
-	})
-
-	addBluetoothDevices("hims", lambda m: any(m.id.startswith(prefix) for prefix in (
-		"BrailleSense",
-		"BrailleEDGE",
-		"SmartBeetle",
-	)))
-
-	# NattiqBraille
-	addUsbDevices("nattiqbraille", KEY_SERIAL, {
-		"VID_2341&PID_8036",  # Atmel-based USB Serial for Nattiq nBraille
-	})
-
-	# superBrl
-	addUsbDevices("superBrl", KEY_SERIAL, {
-		"VID_10C4&PID_EA60",  # SuperBraille 3.2
-	})
-
-	# seika
-	addUsbDevices("seikantk", KEY_HID, {
-		"VID_10C4&PID_EA80",  # Seika Notetaker
-	})
-
-	from brailleDisplayDrivers.seikantk import isSeikaBluetoothDeviceMatch
-	addBluetoothDevices(
-		"seikantk",
-		isSeikaBluetoothDeviceMatch
-	)
-
-	# albatross
-	addUsbDevices("albatross", KEY_SERIAL, {
-		"VID_0403&PID_6001",  # Caiku Albatross 46/80
-	})
+	for display in getSupportedBrailleDisplayDrivers():
+		display.registerAutomaticDetection(DriverRegistrar(display.name))
+	# Hack, Caiku Albatross detection conflicts with other drivers
+	# when it isn't the last driver in the detection logic.
+	_driverDevices.move_to_end("albatross")

```

Comment:
can this be removed now that #15269 has been merged?

Reply:
I'm afraid not. Nothing has changed in this area, i.e. placing albatross first will still mean a major slowdown for other devices as the driver takes so long to decide whether the display is valid.